lib/cli.js

```diff
@@ -24,21 +25,46 @@ module.exports = function broccoliCLI(args) {
     .option('--port <port>', 'the port to bind to [4200]', 4200)
     .option('--host <host>', 'the host to bind to [localhost]', 'localhost')
     .option('--brocfile-path <path>', 'the path to brocfile')
+    .option('--output-path <path>', 'the path to target output folder')
     .action(options => {
       actionPerformed = true;
 
+      const outputDir = options.outputPath;
       const builder = getBuilder({ brocfilePath: options.brocfilePath });
+      const watcher = new Watcher(builder);
+
+      if (outputDir) {
+        if (fs.existsSync(outputDir)) {
+          console.error(outputDir + '/ already exists; we cannot build into an existing directory');
+          process.exit(1);
+        }
+
+        watcher.on('buildSuccess', function() {
+          rimraf.sync(outputDir);
```

Comment:
often people symlink to this directory, deleting it will break that.

in ember-cli we use: https://www.npmjs.com/package/tree-sync for relatively efficient syncing. Thoughts?

Reply:
have some projects that also symlink to output directory and they work fine with serve, one example of Firefox extension: https://github.com/cliqz-oss/browser-core/blob/master/fern/commands/serve.js#L59

Some watches or build system can get confused, I've seen it with `gulp`, but `webpack` in contrary is fine with symlink that get deleted and recreated. 

Did not work with `tree-sync` yet, but looks as optimal solution to the problem (no need for full copy if only some files were changed). 

Should I try to introduce it in this PR?

Reply:
> Should I try to introduce it in this PR?

Ya. We needed to add this to ember-cli, as some users had very large output dirs, (videos/pictures and stuff) and switching to tree-sync fix some serious rebuild issues.

Reply:
done. going to use `tree-sync` in my other projects, it works great.